### ISSUE TITLE
Creating new users

### ISSUE THREAD

**Issue by jtreeves** [jtreeves/read-this-next-ejs#1]
- [ ] Creating a new user crashes Nodemon
```
Executing (05f270c1-8153-4daf-81d3-9242146bb0b4): START TRANSACTION;
Executing (05f270c1-8153-4daf-81d3-9242146bb0b4): SELECT "id", "name", "email", "password", "createdAt", "updatedAt" FROM "users" AS "user" WHERE "user"."email" = 'john@gmail.com' LIMIT 1;
Executing (05f270c1-8153-4daf-81d3-9242146bb0b4): CREATE OR REPLACE FUNCTION pg_temp.testfunc(OUT response "users", OUT sequelize_caught_exception text) RETURNS RECORD AS $func_79c36356f913439cb388692e75d39070$ BEGIN INSERT INTO "users" ("id","name","email","password","createdAt","updatedAt") VALUES (DEFAULT,'John','john@gmail.com','abc1234','2020-12-02 18:38:56.910 +00:00','2020-12-02 18:38:56.910 +00:00') RETURNING * INTO response; EXCEPTION WHEN unique_violation THEN GET STACKED DIAGNOSTICS sequelize_caught_exception = PG_EXCEPTION_DETAIL; END $func_79c36356f913439cb388692e75d39070$ LANGUAGE plpgsql; SELECT (testfunc.response)."id", (testfunc.response)."name", (testfunc.response)."email", (testfunc.response)."password", (testfunc.response)."createdAt", (testfunc.response)."updatedAt", testfunc.sequelize_caught_exception FROM pg_temp.testfunc(); DROP FUNCTION IF EXISTS pg_temp.testfunc();
Executing (05f270c1-8153-4daf-81d3-9242146bb0b4): COMMIT;
John was created
Executing (default): SELECT "id", "name", "email", "password", "createdAt", "updatedAt" FROM "users" AS "user" WHERE "user"."email" = 'john@gmail.com' LIMIT 1;
/Users/jtreeves/Desktop/SEI1019/unit_two/project/read-this-next/node_modules/passport/lib/middleware/authenticate.js:361
        next(err);
        ^

TypeError: next is not a function
    at Strategy.strategy.error (/Users/jtreeves/Desktop/SEI1019/unit_two/project/read-this-next/node_modules/passport/lib/middleware/authenticate.js:361:9)
    at verified (/Users/jtreeves/Desktop/SEI1019/unit_two/project/read-this-next/node_modules/passport-local/lib/strategy.js:81:28)
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
```
- [ ] Sequelize will add the new user to PSQL, but the password information is unhashed

![Screen Shot 2020-12-02 at 1 58 19 PM](https://user-images.githubusercontent.com/15839768/100918418-71990000-34a6-11eb-96cb-8049c0c32044.png)

- [ ] I just tried logging in using the fake user I was able to create, and it gave me this error:
```
TypeError: user.validPassword is not a function
    at /Users/jtreeves/Desktop/SEI1019/unit_two/project/read-this-next/config/ppConfig.js:30:28
    at processTicksAndRejections (node:internal/process/task_queues:93:5)
```

Everything worked fine with creating users when I set up my authentication template during the codealong (passwords were hashed, and nothing crashed). I'm not sure what glitch has crept in (or how), and I'm not sure if it's safe for me to try to edit the node_modules/passport/lib/middleware/authenticate.js file myself. Also my ppConfig file for this project looks exactly like my ppConfig file in my authentication template.

**Comment by jtreeves** [comment]
I just checked my user model file, and it looks different than the one I did for the template. I'm going to try updating that file and remigrating it to see if that fixes the problem.

**Comment by jtreeves** [comment]
Okay, just updated my user model file, and that seems to be helping, but now I'm running into new issues. It says bcrypt is undefined. I'm going to go back to the authentication template and try to gradually add in just what I need, but I'm not sure why the template didn't work right the first time.

**Comment by jtreeves** [comment]
I forgot to add the `const bcrypt = require('bcrypt')` line at the top of the model. All fixed!

**Comment by jtreeves** [comment]
I think I know what I did: When I created the project, I thought I had to recreate a new config file in order to change the name of the database. When I did that, it must have deleted the users file that I had imported from the template, and the new one was created when I used Sequelize to create a users model for my project. In hindsight, I shouldn't have created a new config file; I just needed to change the database's name in that file, then used Sequelize to create the database.
